arbor/include/arbor/load_balance.hpp

```diff
@@ -7,11 +7,54 @@
 namespace arb {
 
 struct partition_hint {
-    constexpr static std::size_t max_size = -1;
 
-    std::size_t cpu_group_size = 1;
-    std::size_t gpu_group_size = max_size;
-    bool prefer_gpu = true;
+    partition_hint() :  cpu_group_size{cpu_group_size_default},
```

Comment:
The reason that these changes were made here was to simplify the Python wrapper?

Instead, I think that this interface should be left as it is (it is much simpler), and the Python wrapper might have to use a shim to implement the partition hint.

Reply:
Changed it back and included exception handling in `partition_load_balance`